src/crimson/os/seastore/cached_extent.h

```diff
@@ -516,21 +540,38 @@ class CachedExtent : public boost::intrusive_ref_counter<
 
 protected:
   CachedExtent(CachedExtent &&other) = delete;
-  CachedExtent(ceph::bufferptr &&ptr) : ptr(std::move(ptr)) {}
+  CachedExtent(std::optional<ceph::bufferptr> &&ptr) : ptr(std::move(ptr)) {
+    assert(this->ptr.has_value());
+    length = this->ptr->length();
+  }
+
+  /// no buffer extent, for lazy read
```

Comment:
Perhaps the extent state in this case would be CLEAN_INCOMPLETE.

Reply:
yes indeed, will do this later.

Reply:
> Perhaps the extent state in this case would be CLEAN_INCOMPLETE.

Probably not worthing it after a second thought, because we should also introduce CLEAN_EXISTING_INCOMPLETE in that case.